var/spack/repos/builtin/packages/chapel/fix_chpl_shared_lib_path.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/tools/chapel-py/setup.py b/tools/chapel-py/setup.py
+index bee452790c..09c16fbda3 100644
+--- a/tools/chapel-py/setup.py
++++ b/tools/chapel-py/setup.py
+@@ -25,6 +25,7 @@ import sys
+ import glob
+
+ chpl_home = str(os.getenv("CHPL_HOME"))
++chpl_lib_install = str(os.getenv("CHPL_SPACK_LIB_INSTALL"))
```

Comment:
@arezaii Thinking more about this...
**Why is this problem Spack-specific?**

IIUC The value being passed in `CHPL_SPACK_LIB_INSTALL` is just `$prefix/lib/chapel/$ver/compiler`, where `$prefix` is exactly the install prefix that was passed at Chapel configure time as `configure --prefix=$prefix`. Two observations:

1. We should not need a new `$CHPL_SPACK_LIB_INSTALL` variable to "smuggle in" this pathname. Isn't Chapel's setup.py capable of fetching the install prefix itself? IIUC this lives in `$CHPL_HOME/configured-prefix`
2. Why is the need for this RPATH entry specific to Spack driving the build/install? Does this reflect a defect in the Chapel installer that should really be eventually fixed upstream?


IOW shouldn't all builds of Chapel always be embedding that directory in the RPATH, to avoid similar problems for stand-alone installs that don't involve Spack?

Reply:
> Why is this problem Spack-specific?

It is not. There is a utility script `fixpaths.py` that does a similar rpath update. This is something that will need to be fixed upstream in the build system for the python bindings.